### PR TITLE
Switch from in-tree mermaid client to out-of-tree manager client

### LINKED ISSUE #1701: Manager client needs to be updated
### What should the feature do?

The manager client needs to be updated, as the current one is missing restore-related methods.

### What is the use case behind this feature?

Test the manual restore procedure in our e2e suite (https://github.com/scylladb/scylla-operator/issues/1671).

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:**
The manager client needs to be updated, as the current one is missing restore-related methods. The restore-related methods are required for https://github.com/scylladb/scylla-operator/issues/1671.
Instead of updating the in-tree mermaid client, this PR removes the in-tree dependency and switches to out-of-tree manager client vendored from Scylla Manager repository.

**Which issue is resolved by this Pull Request:**
Resolves #1701 

/priority important-soon
/kind feature
/area dependency
/cc

### CODE REVIEW

**Comment by rzetelskik** [comment]
https://prow.scylla-operator.scylladb.com/view/gs/scylla-operator-prow/pr-logs/pull/scylladb_scylla-operator/1702/pull-scylla-operator-master-e2e-gke-parallel-clusterip/1750114355044683777#1:test-build-log.txt%3A1351
```shell
  [FAILED] Expected
      <[]*models.TaskListItem | len:0, cap:0>: nil
  to have length 1
  In [It] at: github.com/scylladb/scylla-operator/test/e2e/set/scyllacluster/scyllamanager.go:99 @ 01/24/24 11:36:39.531
  Full Stack Trace
    github.com/scylladb/scylla-operator/test/e2e/set/scyllacluster.glob..func18.1()
    	github.com/scylladb/scylla-operator/test/e2e/set/scyllacluster/scyllamanager.go:99 +0xcf6
```
Looking into what caused this

**Comment by rzetelskik** [comment]
https://github.com/scylladb/scylla-operator/issues/1478#issuecomment-1910075162
/retest

**Comment by rzetelskik** [comment]
The issue was caused by me initially missing the [`nowSafety`](https://github.com/scylladb/scylla-operator/pull/1702/commits/3073f9d6de62f3b1c4101c59cb59e7415f927cc2#diff-0180809a1db5cac9409f3932796fc7b311f20a4011e9c4c6efd167c26afdb273R23) parameter. This is outdated and somewhat broken, but I'll address it separately. This PR is keeping it as it's only supposed to migrate the client out of tree and update it.

File: pkg/controller/manager/sync_action.go [hunk 1]
```diff
@@ -488,3 +489,15 @@ func (a *updateTaskAction) Execute(ctx context.Context, client *mermaidclient.Cl
...
+func messageOf(err error) string {
+	err = errors.Cause(err)
+	switch v := err.(type) {
+	case interface {
+		GetPayload() *models.ErrorResponse
+	}:
+		return v.GetPayload().Message
+	}
+	return err.Error()
+}
```
Comment (zimnx) [inline pkg/controller/manager/sync_action.go:503]: why it's moved here? feels like better place is managerclient

File: pkg/controller/manager/types_old.go [hunk 2]
```diff
@@ -177,3 +186,60 @@ func unescapeFilters(strs []string) []string {
 	}
 	return strs
 }
+
+// parseStartDate parses the supplied string as a strfmt.DateTime.
+func parseStartDate(value string) (strfmt.DateTime, error) {
```
Comment (zimnx) [inline pkg/controller/manager/types_old.go:191]: ditto, why manager logic interpreting start date is part of user package?

File: pkg/controller/manager/types_old.go [hunk 3]
```diff
@@ -177,3 +186,60 @@ func unescapeFilters(strs []string) []string {
...
+
+var (
+	byteCountRe          = regexp.MustCompile(`([0-9]+(?:\.[0-9]+)?)(B|[KMGTPE]iB)`)
+	byteCountReValueIdx  = 1
+	byteCountReSuffixIdx = 2
+)
+
+// parseByteCount returns byte count parsed from input string.
+// This is opposite of StringByteCount function.
+func parseByteCount(s string) (int64, error) {
```
Comment (zimnx) [inline pkg/controller/manager/types_old.go:222]: ditto

File: pkg/controller/manager/sync_action.go (hunk 1 above)
Comment (rzetelskik) [reply pkg/controller/manager/sync_action.go:503]: > feels like better place is managerclient

what do you mean? managerclient is a vendored package

File: pkg/controller/manager/types_old.go (hunk 2 above)
Comment (rzetelskik) [reply pkg/controller/manager/types_old.go:191]: it's not related to manager really - it used to be that way, but it was moved to sctool. https://github.com/scylladb/scylla-manager/blob/d96e5d8338dd0afa5e130ee8cbb88b9cb11bc94b/pkg/command/flag/type.go#L102. This is essentially parsing a value from ScyllaCluster API, which has the same syntax

File: pkg/controller/manager/types_old.go (hunk 3 above)
Comment (rzetelskik) [reply pkg/controller/manager/types_old.go:222]: again, moved away from manager https://github.com/scylladb/scylla-manager/commit/edbbf605148b9f765cc79e72def5e69c69683377, still part of our API though

File: pkg/controller/manager/types_old.go (hunk 2 above)
Comment (zimnx) [reply pkg/controller/manager/types_old.go:191]: I see, I though it's part of Scylla Manager API, but looks like these date and size extensions are part of CLI logic.

File: pkg/controller/manager/sync_action.go (hunk 1 above)
Comment (zimnx) [reply pkg/controller/manager/sync_action.go:503]: right, my bad. I forgot it's vendored now.

**Review by zimnx — APPROVED** [review]
/approve
/assign tnozicka

**Review by tnozicka — APPROVED** [review]
/approve
/lgtm
thanks

**Pull request merged**